### PR TITLE
Create event handler for devices.

### PR DESCRIPTION
**Pull request by TuoMel**

### CODE REVIEW

File: api/services/deviceService.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ const initializeDevice = async (name, country) => {
                 return {success: false, message: "Country not found or not supported!"};
             }
 
-            const result = await Devices.create({name: name, uuid: uuidCreator.v4(), country: countryCode});
+            const result = await Devices.create({name: name, uuid: uuidCreator.v4(), country: countryCode, ...(eventsupport && {eventsupport: eventsupport})});
```
Comment (TuoMel) [inline api/services/deviceService.js:31]: Modify this

File: api/controllers/apicontroller.js [hunk 2]
```diff
@@ -617,6 +616,69 @@ const getDailyAverages = async (req, res) => {
...
+            message: "New event created successfully!",
+            data: result.data
+        });
+    } catch (error) {
+        return res.status(500).json({
+            message: error.message || 'Internal Server Error'
+        });
+    }
+}
+
```
Comment (TuoMel) [inline api/controllers/apicontroller.js:675]: Too many spaces.

File: api/services/eventService.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            return {success: false, message: "Subscriber already exists!"}
+        }
+
+        const deviceResult = await Devices.findByPk(device);
+
+        if (!deviceResult) {
+            return {success: false, message: "Device not found!"}
+        }
+
+        ip = "172.168.1.1";
```
Comment (TuoMel) [inline api/services/eventService.js:18]: This needs to be removed for production

File: services/eventService.js [hunk 4]
```diff
@@ -34,6 +37,33 @@ const sendDailyWeatherEmail = async () => {
 }
 
 
+const checkForEvents = async (device, measurement) => {
+    try {
+        const events = await getEventsForDevice(device).then(result => result.data);
+
```
Comment (TuoMel) [inline services/eventService.js:43]: use early return here

Comment (TuoMel) [reply services/eventService.js:43]: Done

File: api/controllers/apicontroller.js (hunk 2 above)
Comment (TuoMel) [reply api/controllers/apicontroller.js:675]: Done

File: api/services/deviceService.js (hunk 1 above)
Comment (TuoMel) [reply api/services/deviceService.js:31]: Done

**Comment by TuoMel** [comment]
DONE! Please, review.

**Review by landyboi — APPROVED** [review]

**Pull request merged**